modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/sourcer/object/ObjectSourcerImpl.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.builder.internal.sourcer.object;
+
+import com.liferay.headless.builder.internal.sourcer.api.PropertyInfo;
+import com.liferay.headless.builder.internal.sourcer.api.Sourcer;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.rest.dto.v1_0.ObjectEntry;
+import com.liferay.object.rest.manager.v1_0.DefaultObjectEntryManager;
+import com.liferay.object.rest.manager.v1_0.ObjectEntryManagerRegistry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterContext;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.yaml.openapi.FieldDefinition;
+import com.liferay.portal.vulcan.yaml.openapi.Schema;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.core.UriInfo;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(service = Sourcer.class)
+public class ObjectSourcerImpl implements Sourcer {
+
+	@Override
+	public Map<String, PropertyInfo> getPropertiesInfo(
+		String entityName, Map<String, Schema> propertySchemas) {
+
+		Map<String, PropertyInfo> stringObjectPropertyMap = new HashMap<>();
+
+		for (Map.Entry<String, Schema> schemaEntry :
+				propertySchemas.entrySet()) {
+
+			Schema value = schemaEntry.getValue();
+
+			FieldDefinition fieldDefinition = value.getFieldDefinition();
+
+			String objectFieldName = fieldDefinition.getName();
+
+			stringObjectPropertyMap.put(
+				schemaEntry.getKey(),
+				new PropertyInfo(
+					entityName, objectFieldName, schemaEntry.getKey()));
+		}
+
+		return stringObjectPropertyMap;
+	}
+
+	@Override
+	public Object getValue(
+			PropertyInfo propertyInfo, Object pathParameterValue,
+			HttpServletRequest httpServletRequest, UriInfo uriInfo)
+		throws Exception {
+
+		long objectEntryId = 0;
+
+		if (pathParameterValue instanceof Long) {
+			objectEntryId = (long)pathParameterValue;
+		}
+
+		String internalClass = propertyInfo.getInternalClass();
+
+		long objectDefinitionId = GetterUtil.getLong(
+			internalClass.split("#")[1]);
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionId);
+
+		DefaultObjectEntryManager defaultObjectEntryManager =
+			(DefaultObjectEntryManager)
```

Comment:
Maybe use [DefaultObjectEntryManagerProvider.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-rest-api/src/main/java/com/liferay/object/rest/manager/v1_0/DefaultObjectEntryManagerProvider.java#L22), in case they add more logic to the provider, so we don't have to fix our code separately.

Reply:
True! I didn't see that method 😅 Thanks! I've just changed it